### PR TITLE
added last.fm module

### PR DESCRIPTION
**Pull request by Krishna-Singhal**

### CODE REVIEW

File: plugins/lastfm.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+        out += f'    __● SongName:__ `{user.get_now_playing()}`'
+    else:
+        out += '__NowPlaying:__ `False`'
+
+    if user.get_image():
+        path = os.path.join(Config.DOWN_PATH, f"{user.get_name()}.png")
+        if not os.path.exists(path):
+            await pool.run_in_thread(wget.download)(user.get_image(), path)
+    else:
+        path = "resources/no_image.png"
```
Comment (Phyco-Ninja) [inline plugins/lastfm.py:124]: maybe you forgot to upload this media

File: plugins/lastfm.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+        if NOW_PLAYING[1] != song.get_name():
+            NOW_PLAYING[1] = song.get_name()
+            for chat_id in CHAT_IDS:
+                out = f"{USERNAME} __is currently Listening to:__\n\n"
+                k = get_track_info(song)
+                if not k:
+                    NOW_PLAYING[0] = False
+                    return
+                out += k
+                if userge.has_bot:
```
Comment (Phyco-Ninja) [inline plugins/lastfm.py:222]: is this only for Dual Mode users? 🤔

File: plugins/lastfm.py [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+        await msg.edit("`Auto updates Stopped.`")
+    else:
+        track = await LastFm(msg).now_playing()
+        if not track:
+            return await msg.edit("`Currently you are not listening to any Song...`")
+        if NOW_PLAYING[1] != track.get_name():
+            NOW_PLAYING[1] = track.get_name()
+
+        out = f"{USERNAME} __is currently Listening to:__\n\n"
+        k = get_track_info(track)
```
Comment (Phyco-Ninja) [inline plugins/lastfm.py:81]: forgot to add k to out? or it is intentional

File: plugins/lastfm.py [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+            await msg.edit(
+                "`This plugins needs environmental variables,"
+                " For more info see` "
+                "[this post](https://t.me/UnofficialPluginsHelp/123).",
+                disable_web_page_preview=True
+            )
+    return checker
+
+
+@check_creds
```
Comment (Phyco-Ninja) [inline plugins/lastfm.py:51]: btw does it work? (~_^)

Comment (Krishna-Singhal) [reply plugins/lastfm.py:51]: yep I tried it

File: plugins/lastfm.py [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+        return out
+
+    async def get_last_played(self) -> None:
+        await self.msg.edit("`getting tracks...`")
+        limit = self.msg.flags.get('-l', 10)
+        if self.msg.filtered_input_str:
+            user = self.get_user(self.msg.filtered_input_str)
+        else:
+            user = self.get_user()
+        out = f"Last played tracks of {user.get_name}:\n"
```
Comment (Phyco-Ninja) [inline plugins/lastfm.py:372]: change user.get_name to user.get_name() and try int() on limit.

**Comment by Phyco-Ninja** [comment]
seems okay now.-_-+

**Comment by Krishna-Singhal** [comment]
> seems okay now.-_-+

yep thanks.

**Pull request merged**